src/smr/smr_types.rs

```diff
@@ -57,26 +65,39 @@ impl Default for Step {
 impl Into<u8> for Step {
     fn into(self) -> u8 {
         match self {
-            Step::Propose => 0,
-            Step::Prevote => 1,
-            Step::Precommit => 2,
-            Step::Commit => 3,
+            Step::Brake => 0,
+            Step::Propose => 1,
+            Step::Prevote => 2,
+            Step::Precommit => 3,
+            Step::Commit => 4,
         }
     }
 }
 
 impl From<u8> for Step {
     fn from(s: u8) -> Self {
         match s {
-            0 => Step::Propose,
-            1 => Step::Prevote,
-            2 => Step::Precommit,
-            3 => Step::Commit,
+            0 => Step::Brake,
+            1 => Step::Propose,
+            2 => Step::Prevote,
+            3 => Step::Precommit,
+            4 => Step::Commit,
             _ => panic!("Invalid vote type!"),
         }
     }
 }
 
+///
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+pub enum FromWhere {
+    ///
+    PrevoteQC(u64),
```

Comment:
should involve aggregate signature for verifying, and the two below are the same